File: php/src/MoneyProblem/Domain/Money.php
<?php

namespace MoneyProblem\Domain;

class Money {
    private $amount;
    private Currency $currency;
    
    public function __construct($amount,Currency $currency)
    {
        $this->amount = $amount;
        $this->currency = $currency;
    }

    public function times($times){
        $temp = $this->amount * $times;
        $this->setAmount($temp);
        return $temp;
    }
    
    public function getAmount(){
        return $this->amount;
    }
    public function setAmount($amount){
        $this->amount=$amount;
    }

    public function getCurrency():Currency{
        return $this->currency;
    }
}

File: java/src/main/java/money_problem/domain/Money.java
package money_problem.domain;

public class Money {
    Currency currency;
    double amount;
    
    Money(Currency myCurrency, double myAmount){
        currency = myCurrency;
        amount = myAmount;
    }

    public Currency getCurrency(){
        return this.currency;
    }

    public double getAmount(){
        return this.amount;
    }

    public void addCurrency(double myAmount){
        this.amount += myAmount;
    }
}
